DotNet Libraries.txt

```diff
@@ -0,0 +1,272 @@
+      EXIT(GetTokenDateTimeValue(TempStandardText,'EXP') < CURRENTDATETIME);
+    END;
+
+    PROCEDURE GetTokenValue@10017290(VAR TempStandardText@10017293 : TEMPORARY Record 7;Id@10017291 : Text) : Text;
+    BEGIN
+      IF TempStandardText.GET(Id) THEN
+        EXIT(TempStandardText.Description);
+    END;
+
+    PROCEDURE GetTokenDateTimeValue@10017292(VAR TempStandardText@10017292 : TEMPORARY Record 7;Id@10017291 : Text) : DateTime;
+    VAR
+      Timestamp@10017293 : Decimal;
+    BEGIN
+      IF NOT EVALUATE(Timestamp,GetTokenValue(TempStandardText,Id)) THEN EXIT;
+      EXIT(EvaluateUnixTimestamp(Timestamp));
+    END;
+
+    PROCEDURE GetTokenDetails@10017293(AccessToken@10017294 : Text;VAR TempStandardText@10017290 : TEMPORARY Record 7);
+    VAR
+      JwtSecurityTokenHandler@10017293 : DotNet "'System.IdentityModel.Tokens.Jwt, Version=5.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.System.IdentityModel.Tokens.Jwt.JwtSecurityTokenHandler";
+      JwtSecurityToken@10017292 : DotNet "'System.IdentityModel.Tokens.Jwt, Version=5.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'.System.IdentityModel.Tokens.Jwt.JwtSecurityToken";
+      Claim@10017295 : DotNet "'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.Security.Claims.Claim";
+    BEGIN
+      IF AccessToken = '' THEN EXIT;
+      JwtSecurityTokenHandler := JwtSecurityTokenHandler.JwtSecurityTokenHandler;
+      JwtSecurityToken := JwtSecurityTokenHandler.ReadToken(AccessToken);
+
+      WITH TempStandardText DO
+        FOREACH Claim IN JwtSecurityToken.Claims DO BEGIN
+          Code := Claim.Type;
+          Description := Claim.Value;
+          INSERT;
+        END;
+    END;
+
+    LOCAL PROCEDURE EvaluateUnixTimestamp@19(Timestamp@1002 : BigInteger) : DateTime;
+    VAR
+      ResultDateTime@1000 : DateTime;
+      EpochDateTime@1001 : DateTime;
+      TimezoneOffset@1003 : Duration;
+      TimestampInMilliseconds@1004 : BigInteger;
+    BEGIN
+      IF NOT GetUserTimezoneOffset(TimezoneOffset) THEN
+        TimezoneOffset := 0;
+
+      EpochDateTime := CREATEDATETIME(DMY2DATE(1,1,1970),0T);
+      TimestampInMilliseconds := Timestamp * 1000;
+      ResultDateTime := EpochDateTime + TimestampInMilliseconds + TimezoneOffset;
+      EXIT(ResultDateTime);
+    END;
+
+    [TryFunction]
+    LOCAL PROCEDURE GetUserTimezoneOffset@5(VAR Duration@1001 : Duration);
+    VAR
+      UserPersonalization@1004 : Record 2000000073;
+      TimeZoneInfo@1003 : DotNet "'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.TimeZoneInfo";
+      TimeZone@1000 : Text;
+    BEGIN
+      UserPersonalization.GET(USERSECURITYID);
+      TimeZone := UserPersonalization."Time Zone";
+      TimeZoneInfo := TimeZoneInfo.FindSystemTimeZoneById(TimeZone);
+      Duration := TimeZoneInfo.BaseUtcOffset;
+    END;
+
+    BEGIN
+    END.
+  }
+}
+
+OBJECT Codeunit 55986 Json/Xml Conversion Library

```

Comment:
Did you look at codeunit 1237 which does much of the same?

Reply:
Yes, it does but only from json to xml.  I need both directions.  Having these functions will allow me to use XMLPorts for json files - until XMLPorts become DataPorts that will support json natively.

Reply:
Example here: http://www.dynamics.is/?p=2637